.github/workflows/Dockerfile

```
@@ -1,17 +1,19 @@
+ARG BASE_OS
+ARG BASE_DISTRO
 ARG BASE_IMAGE
 ARG K_VERSION
 ARG Z3_VERSION
-FROM z3:${BASE_IMAGE}-${Z3_VERSION} as Z3
+FROM z3:${BASE_DISTRO}-${Z3_VERSION} as Z3
 
-ARG BASE_IMAGE
+ARG BASE_OS
```

```suggestion
ARG BASE_OS
ARG BASE_DISTRO
```

You have to repeat all arguments before every `FROM`.